receive.go

```diff
@@ -139,11 +143,11 @@ func (r *receiver) run(ctx context.Context) error {
 				r.conn.SendMsg(&types.Packet{Type: types.PACKET_ERR, Data: []byte(retErr.Error())})
 			}
 		}()
-		destWalker := emptyWalker
-		if !r.merge {
-			destWalker = getWalkerFn(r.dest)
+		if r.merge {
+			err = createDir(ctx, dw.HandleChange, w.fill)
+		} else {
+			err = doubleWalkDiff(ctx, dw.HandleChange, getWalkerFn(r.dests[0]), w.fill, r.filter, r.differ)
```

Comment:
Not sure how this works if the comparison for the differ is always based on the first path?